tests/storage/upgrade/test_upgrade_storage.py

```diff
@@ -164,19 +188,31 @@ def test_vm_snapshot_restore_check_after_upgrade(
         ],
         scope=DEPENDENCY_SCOPE_SESSION,
     )
-    def test_vm_snapshot_restore_create_after_upgrade(self, cirros_vm_for_upgrade_b, snapshots_for_upgrade_b):
+    def test_vm_snapshot_restore_create_after_upgrade(self, rhel_vm_for_upgrade_b, snapshots_for_upgrade_b):
```

Comment:
aren't there dedicated restore tests that run as part of post_upgrade? why is this test needed here?

Reply:
We write file1, do the snapshot, write file2 --> upgrade --> restore from a pre-upgrade snapshot, verify only file1 is there. This is to verify that pre-upgrade snapshot is not broken (we had a bug when snapshot was impossible to restore from after OCP upgrade because of api version change). Regular port-upgrade tests couldn't catch it.